pkg/mbset/mbset.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package mbset
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/api/processapi"
+	"github.com/cilium/tetragon/pkg/bpf"
+)
+
+const (
+	MapName   = "tg_mbset_map"
+	InvalidID = ^uint32(0)
+	MaxIDs    = 64 // this value should correspond to the number of bits we can fit in mbset_t
+)
+
+type bitSet = uint64
+
+func openMap() (*ebpf.Map, error) {
+	fname := filepath.Join(bpf.MapPrefixPath(), MapName)
+	ret, err := ebpf.LoadPinnedMap(fname, &ebpf.LoadPinOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", fname, err)
+	}
+	return ret, nil
+}
+
+type State struct {
+	mu       sync.Mutex
+	nextID   uint32
+	mbsetMap *ebpf.Map
+}
+
+func newState() (*State, error) {
+	m, err := openMap()
+	if err != nil {
+		return nil, fmt.Errorf("failed to open map: %w", err)
+	}
+	return &State{
+		mbsetMap: m,
+	}, nil
+}
+
+func (s *State) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	return s.mbsetMap.Close()
+}
+
+// AllocID allocates a new ID
+func (s *State) AllocID() (uint32, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if s.nextID >= MaxIDs {
```

Comment:
should we have some ID cleanup logic, because IIUC this will effectively fail to add policy with followChildren, when we cross the MaxIDs value, which is 64 now... we could reach that after 64 attempts to load policy through tetra interface right?

Reply:
Indeed we need cleanup logic to go over 64 policies but I left it as a followup since it's somewhat tricky. We can do it by introducing a lock to update clean up the bitset, but I'm trying to figure out if there is also a better way by passing the map updates to kernel side so that they can happen without a lock.

Reply:
ok, we have a sensor removal hook, maybe we could reclaim those ids from there

Reply:
Yap, the tricky part is cleaning the bits from the processes in `execve_map`.